packages/edit-site/src/components/global-styles/screen-color-element.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { __experimentalColorGradientControl as ColorGradientControl } from '@wordpress/block-editor';
+/**
+ * Internal dependencies
+ */
+import ScreenHeader from './header';
+import {
+	getSupportedGlobalStylesPanels,
+	useSetting,
+	useStyle,
+	useColorsPerOrigin,
+} from './hooks';
+
+const elements = {
+	text: {
+		description: __(
+			'Set the default color used for text across the site.'
+		),
+		title: __( 'Text' ),
+	},
+	caption: {
+		description: __( 'Set the colors used for captions across the site' ),
+		title: __( 'Captions' ),
+	},
+	button: {
+		description: __(
+			'Set the default colors used for buttons across the site.'
+		),
+		title: __( 'Buttons' ),
+	},
+};
+
+function ScreenColorElement( { name, element } ) {
+	const supports = getSupportedGlobalStylesPanels( name );
+	const colorsPerOrigin = useColorsPerOrigin( name );
+	const [ solids ] = useSetting( 'color.palette', name );
+	const [ areCustomSolidsEnabled ] = useSetting( 'color.custom', name );
+	let [ isBackgroundEnabled ] = useSetting( 'color.background', name );
+	const [ isTextEnabled ] = useSetting( 'color.text', name );
+
+	let textColorElementSelector = 'elements.' + element + '.color.text';
+	const backgroundColorElementSelector =
+		'elements.' + element + '.color.background';
+
+	let hasElementColor = false;
+
+	switch ( element ) {
+		case 'button':
+			hasElementColor =
+				supports.includes( 'buttonColor' ) &&
+				isBackgroundEnabled &&
+				( solids.length > 0 || areCustomSolidsEnabled );
+			break;
+		case 'text':
+			hasElementColor =
+				supports.includes( 'color' ) &&
+				isTextEnabled &&
+				( solids.length > 0 || areCustomSolidsEnabled );
+			textColorElementSelector = 'color.text';
+			isBackgroundEnabled = false;
+			break;
+
+		default:
+			hasElementColor =
+				supports.includes( 'color' ) &&
+				( solids.length > 0 || areCustomSolidsEnabled );
+			break;
+	}
```

Comment:
This is potentially a little nit-picky, and it doesn't have any impact on the overall functioning of this code, so feel free to ignore this comment!

Some of the global styles files get complex pretty quickly as we add additional elements and controls, and looking at the changes here made me think of a couple of small changes that might improve readability. Is there an opportunity here to add a `useHasElementColor` hook to this file, similar to how the `border-panel.js` and `dimensions-panel.js` files have a bunch of `useHas` hooks?

We could then potentially swap out the switch statement to use that hook, where the internals might be a shorter series of if statements:

```js
	if ( solids.length > 0 || areCustomSolidsEnabled ) {
		if ( element === 'button' ) {
			hasElementColor =
				supports.includes( 'buttonColor' ) && isBackgroundEnabled;
		} else if ( element === 'text' ) {
			hasElementColor = supports.includes( 'color' ) && isTextEnabled;
		} else {
			hasElementColor = supports.includes( 'color' );
		}
	}
```

Ideally `ScreenColorElement` would then either get its settings via calling other hooks, or do a look-up to the `elements` object.

My feedback here is very much influenced by my personal preference that I find `switch` statements challenging to read, so definitely not a blocker to this PR landing.